sql/parser/deep_copy_test.go

```diff
@@ -1,371 +0,0 @@
-// Copyright 2016 The Cockroach Authors.
```

Comment:
We don't need the deep copy stuff anymore. It had kind of a faulty semantic ("deep" meant up to Expr leaves but did not deep copy things like `DTuple` or `*DDecimal`). There are existing tests for Walk and Normalize, and I added a check that the original expression is not modified to those.